### PR TITLE
modules: remove file extension and directory resolving for esm

### PR DESCRIPTION
**Pull request by MylesBorins**
this build on top of the limitations of the package name map proposal.
It removes file extensions and directory resolution in the resolver.

This is only currently limited for local development. When resolving
dependencies file extension and directory resolution will still work.

Refs: https://github.com/domenic/package-name-maps

Moved from https://github.com/nodejs/node/pull/21729

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: doc/api/esm.md [hunk 1]
```diff
@@ -54,6 +54,14 @@ property:
 
 ## Notable differences between `import` and `require`
 
+### Mandatory file extensions
+
+You must provide a file extension when using the `import` keyword.
```
Comment (michael-ciniawsky) [inline doc/api/esm.md]: To clearify this does include dynamic imports ?  (`import()`)

**Comment by devsnek** [comment]
i'd like to make sure we have this repo set up to automatically pull updates from nodejs/node before we're merging changes

**Comment by bmeck** [comment]
We have people discussing the continues use of bundlers in https://github.com/nodejs/ecmascript-modules/pull/1 and not focusing on web compatibility since they can continue to use the solutions bundlers provide. If that is the mentality we don't need to do this PR since resolution could be provided by ahead of time tools like bundlers.

Additionally, given the package name maps approach and proof of creating maps that support index files, extension completion, etc. I don't see the need for this PR even with people wanting to use bundlers if they want to manually make package name maps, or using a different tool.

**Comment by MylesBorins** [comment]
Closing in lieu of https://github.com/nodejs/ecmascript-modules/pull/6

**Pull request closed without merging**